src/theory/quantifiers/sygus/sygus_repair_const.h

```diff
@@ -0,0 +1,180 @@
+  Node d_base_inst;
+  /**
+   * whether any sygus type for the candidate variables of d_base_inst (the
+   * syntactic restrictions) allows all constants. If this flag is false, then
+   * this class is a no-op.
+   */
+  bool d_allow_constant_grammar;
+  /** easy quantified logic
+   *
+   * Set to true if the background logic is reasonably easy to answer
+   * quantified queries in. Currently, we say the logic is not easy if
+   * bit-vectors or non-linear arithmetic are enabled.
+   */
+  bool d_easy_quantified_logic;
+  /** map from skeleton variables to first-order variables */
+  std::map<Node, Node> d_sk_to_fo;
+  /** reverse map of d_sk_to_fo */
+  std::map<Node, Node> d_fo_to_sk;
+  /** a cache of satisfiability queries of the form [***] above we have tried */
+  std::unordered_set<Node, NodeHashFunction> d_queries;
+  /**
+   * Register information for sygus type tn, tprocessed stores the set of
+   * already registered types.
+   */
+  void registerSygusType(TypeNode tn, std::map<TypeNode, bool>& tprocessed);
+  /**
+   * Returns true if n is a term of a sygus datatype type that allows all
+   * constants, and n encodes a constant. The term n must of a sygus datatype
+   * type.
+   */
+  bool isRepairableConstant(Node n);
+  /** get skeleton
+   *
+   * Returns a skeleton for n, where the subterms of n that are repairable
+   * constants are replaced by free variables. Since we are interested in
+   * returning canonical skeletons, the free variables we use in this
+   * replacement are taken from TermDbSygus, where we track indices
+   * in free_var_count. Variables we introduce in this way are added to sk_vars.
+   * The mapping sk_vars_to_subs contains entries v -> c, where v is a
+   * variable in sk_vars, and c is the term in n that it replaced.
+   */
+  Node getSkeleton(Node n,
+                   std::map<TypeNode, int>& free_var_count,
+                   std::vector<Node>& sk_vars,
+                   std::map<Node, Node>& sk_vars_to_subs);
+  /** get first-order query
+   *
+   * This function returns a formula that is equivalent to the negation of the
+   * synthesis conjecture, where candidates are replaced by candidate_skeletons,
+   * whose free variables are in the set sk_vars. The returned formula
+   * is a first-order (quantified) formula in the background logic, without UF,
+   * of the form [***] above.
+   */
+  Node getFoQuery(const std::vector<Node>& candidates,
+                  const std::vector<Node>& candidate_skeletons,
+                  const std::vector<Node>& sk_vars);
+  /** fit to logic
+   *
+   * This function ensures that a query of the form [***] above fits the given
+   * logic. In our approach for constant repair, replacing constants by
+   * variables may introduce e.g. non-linearity. If non-linear arithmetic is
+   * not enabled, we must undo some of the variables we introduced when
+   * inferring candidate skeletons.
+   *
+   * This function may remove variables from sk_vars and the map
```

Comment:
mention here that this function is using the one below to check variables must be excluded